tsconfig.json

```diff
@@ -1,44 +1,41 @@
 {
     "compilerOptions": {
         "module": "CommonJS",
-        "lib": ["es2019", "es2020.promise", "es2020.bigint", "es2020.string"],
+        "lib": [
+            "es2019",
+            "es2020.promise",
+            "es2020.bigint",
+            "es2020.string",
+            "dom"
```

Comment:
I'm still a bit confused about this. If I remove "dom" then I get build errors in `Search.ts`, which has references to dom concepts such as "document" and `HTMLScriptElement`.  I assume this has something to do with my exporting `IDocument` out of `Search.ts`, but even when I put everything back I still get errors like this:

```
> esbuild src/lib/output/themes/default/assets/bootstrap.ts --bundle --minify --outfile=static/main.js

X [ERROR] Could not resolve "path"      

    src/lib/models/sources/file.ts:1:22:
      1 │ import * as Path from "path";
        ╵                       ~~~~~~

  The package "path" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "--platform=node" to do that, which will remove this error.
```


I see you have a separate script to build the themes, but if I remove "dom" and move the

Reply:
I think it has something to do with importing `ReflectionKind` into Search--it seems to open cross-ts-checking that is not meant to be across some kind of separation in the themes folder structure that I don't understand.

Reply:
?

Reply:
I reverted the `IDocument` stuff. I'll leave it for someone else to figure out.

Reply:
That error is because the frontend doesn't have access to node modules, and bootstrap.ts was updated to import IDocument from a path which also ended up importing something that depended on native node modules. I'm fine just leaving the any for now, hasn't really caused any major pain. Still need to remove `dom` here though.

Reply:
Whoops, forgot to remove "dom". 26d4a9a7

I have to say I found several aspects of this project a bit confusing:
* the stuff in /themes seems to be silo'ed off into its own space, but the separation is sort of passive and not clear (If you really, *really* don't want it to mix with the library stuff, why not put it in a separate package, and have it import stuff from package scope like any other standalone app?
* what, if anything, checks files like `Search.ts` ? You can type a garbage string right into the middle of the file, and `build`, `build:tsc` and `build:themes` all pass without complaint
* some files under /themes do seem to reach up to get interfaces from the main library code (eg [here](https://github.com/TypeStrong/typedoc/blob/master/src/lib/output/themes/default/DefaultTheme.tsx#L9))

Etc. Overall, I enjoyed exploring the code.